src/lib/session-storage.js

```diff
@@ -0,0 +1,26 @@
+/**
+ * This yields either sessionStorage, or when that particular
+ * API is not available, yields an object with limited API
+ * that behaves _like_ sessionStorage for the duration of
+ * the app, but does not persist across hard reloads.
+ */
+const storage = (() => {
+  try {
+    if (typeof sessionStorage !== "undefined") {
+      return sessionStorage;
+    }
+  }
+
+  catch (e) {
+    // No sessionStorage is currently available.
+    // This is not a true "error" in terms of UX.
+  }
+
+  return {
```

Comment:
does it run this even though it's outside the `catch` block or does this need to be inside `finally`?

Reply:
this runs just fine. If the try throws, catch catches it, and then execution proceeds. In this case we just don't do anything with that `e`

Reply:
On a fun additional note: `finally` has been the source of hilarious bugs, because there isn't actually any guarantee that it'll run. On the web, usually not much a problem, but say you do a payment rollback in `finally` and someone just closes the tab before it can kick in: rofflecopters.